bugsnag-spring/src/main/java/com/bugsnag/SpringBootLoadedCondition.java

```diff
@@ -0,0 +1,22 @@
+package com.bugsnag;
+
+import org.springframework.context.annotation.Condition;
+import org.springframework.context.annotation.ConditionContext;
+import org.springframework.core.io.DefaultResourceLoader;
+import org.springframework.core.io.ResourceLoader;
+import org.springframework.core.type.AnnotatedTypeMetadata;
+
+/**
+ * Check whether spring-boot is available to the application.
+ */
+class SpringBootLoadedCondition implements Condition {
+
+    @Override
+    public boolean matches(ConditionContext context,
+                           AnnotatedTypeMetadata metadata) {
+        ResourceLoader loader = context.getResourceLoader() == null
+                ? new DefaultResourceLoader() : context.getResourceLoader();
+        return loader != null
```

Comment:
The `loader` cannot be null here, and does not appear to be used. I think this method can be simplified to just

```java
return context.getClassLoader().getResource("org/springframework/boot") != null;
```
unless there's some magic going on on the line above. In case of magic, we should probably add a comment explaining what's going on.

This comment applies to other `Condition` implementations as well.